.github/workflows/queasars_deploy_documentation.yml

```diff
@@ -0,0 +1,68 @@
+name: Deploy QUEASARS sphinx documentation
+
+on:
+  # Runs only on tags of the form vX.Y.Z
+  push:
+    tags: ['v[0-9].[0-9].[0-9]']
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+# Sets permissions of the GITHUB_TOKEN to allow deployment to GitHub Pages
+permissions:
+  contents: read
+  pages: write
+  id-token: write
+
+# Allow only one concurrent deployment, skipping runs queued between the run in-progress and latest queued.
+# However, do NOT cancel in-progress runs as we want to allow these production deployments to complete.
+concurrency:
+  group: "pages"
+  cancel-in-progress: false
+
+jobs:
+  # Build job
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Set up Python
+        uses: actions/setup-python@v3
+        with:
+          python-version: "3.11"
+      - name: Install pipx
+        run: |
+          python3 -m pip install --upgrade pip
+          python3 -m pip install --user pipx
+          python3 -m pipx ensurepath
+      - name: Install poetry
+        run: |
+          pipx install poetry==1.7.0
+      - name: Install dependencies
+        run: |
+          poetry install --with docs
+      - name: Setup Pages
+        id: pages
+        uses: actions/configure-pages@v3
+      - name: Build Sphinx documentation
+        run: |
+          cd docs
+          poetry run make clean
+          poetry run make html
+          cd ..
```

Comment:
Is this directory change necessary? poetry is now run from the docs subfolder, is this actually required? That looks slightly strange to me.
Also, if the change back at the end is needed, does this mean that the current working directory is preserved between steps in the job?

Reply:
The reason is that the makefile is in the docs folder.
The alternative would be to use this instead of switching folders:
```poetry run .\docs\make clean``` and
```poetry run .\docs\make html``` Would you prefer this?

Reply:
No, I think it's fine, just for my curiosity.